### PR TITLE
First version of OSS Support

### PR DESCRIPTION
**Pull request by bechamp**
Hi,
This is in response to a convo Adam and I quickly had on slack. Just an announcement of the awareness about the support issues that exist in some areas of open source, specificly in security software. I'm not really affiliated with any of this, just another engineer with concerns. Please, take a look and let me know if this is something you all want to address at all.

Thanks,
Kris

### CODE REVIEW

**Comment by adamstac** [comment]
Hey @bechamp we hadn't forgotten about this, just busy with the first few weeks of going full-time. @jerodsanto and I plan to sync up soon on thoughts and post back here when we can give feedback to you.

Shouldn't be long.

**Comment by bechamp** [comment]
@adamstac, thanks for the update!

File: draft/oss-support/post.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  - [The call to arms from GnuPG][link_gnupg_blog]
+  - [The broader coverage by Slashdot][link_gnupg_slashdot]
+
+## Concerns
+
+GnuPG is only one example of backbone software projects that comprise the core
+ecosystem that is the net. There are many others. A concerning notion is that
+in a time when privacy concerns, brought about by government and overreaching
+commercial entities, is the fact that many secure services are maintained by a
+hand full of developers. These secure services are really at the front lines in
```
Comment (jbranchaud) [inline draft/oss-support/post.md]: `hand full` should be `handful`

File: draft/oss-support/post.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# OSS Support
```
Comment (jerodsanto) [inline draft/oss-support/post.md]: Maybe think of a few alternate titles for this. Something a little more catchy/intriguing?

File: draft/oss-support/post.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+Free software. It's not _really_ free. Time, effort, innovation and labor go
+into developing the open source projects that we use and sometimes can't live
+without. A lot of these projects live in the background, known only to those
```
Comment (jerodsanto) [inline draft/oss-support/post.md]: > A lot of these projects live in the background, known only to those that dig into the low level building blocks of the net or in the lower depths of the application layer of open and closed operating systems

This isn't exactly a _run-on_, but it is a bit wordy. Maybe revisit this section with fresh eyes.

File: draft/oss-support/post.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+that dig into the low level building blocks of the net or in the lower depths
+of the application layer of open and closed operating systems (see the EULA of
+OSX and you'll see open source projects listed). Many open source projects find
+funding through sales of support, licenses, and by hosting. But quite a few are
+just put out there and sometimes they find there way into the tool boxes of
```
Comment (jerodsanto) [inline draft/oss-support/post.md]: > find there way

should be

> find _their_ way

File: draft/oss-support/post.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+devs and designers, to the point where they almost become ubiquitous.
+
+When you think about it, how many of the open source projects that you use all
+the time are not heavily funded operations? VI, Emacs, heck even small tools
+like sed to niche languages like tcl or lua. These are all entrenched and still
```
Comment (jerodsanto) [inline draft/oss-support/post.md]: Please add links to tcl and lua. Also, fin their proper names (capitalization-style) and use those.

File: draft/oss-support/post.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+active. They may receive funding through the Free Software Foundation and
+donations, but when was the last time you paid any money or gave anything back
+to the developers of that editor, that library, that module that you always
+turn too?
```
Comment (jerodsanto) [inline draft/oss-support/post.md]: s/too/to/

File: draft/oss-support/post.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+the time are not heavily funded operations? VI, Emacs, heck even small tools
+like sed to niche languages like tcl or lua. These are all entrenched and still
+active. They may receive funding through the Free Software Foundation and
+donations, but when was the last time you paid any money or gave anything back
+to the developers of that editor, that library, that module that you always
+turn too? 
+
+## Inspiration
+
+This topic was inspired by an event that occurred today, February 5 2015. The
```
Comment (jerodsanto) [inline draft/oss-support/post.md]: Please update this since the moment has passed.

File: draft/oss-support/post.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+_story_ of an open source project was showcased to the world. The showcase was
+not made in a big way, I learned about it through a Slashdot RSS. But when
+GnuPG curator, Werner Koch, published a blog post breaking down the costs of
```
Comment (jerodsanto) [inline draft/oss-support/post.md]: add link to the blog post

File: draft/oss-support/post.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+maintaining that project the response was made in a big way. That blog post was
+made back in December but the word seemed to get out today and in one day over
```
Comment (jerodsanto) [inline draft/oss-support/post.md]: "made" would read better as "written" or "published".

**Comment by bechamp** [comment]
Thanks for the focus guys, I'm heading to family events this weekend but will look on Monday.

File: draft/oss-support/post.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+Here are some related links:
+
+  - [The call to arms from GnuPG][link_gnupg_blog]
+  - [The broader coverage by Slashdot][link_gnupg_slashdot]
+
+## Concerns
+
+GnuPG is only one example of backbone software projects that comprise the core
+ecosystem that is the net. There are many others. A concerning notion is that
+in a time when privacy concerns, brought about by government and overreaching
```
Comment (jerodsanto) [inline draft/oss-support/post.md]: > ... concerns, brought about by government and overreaching commercial entities, is ...

I think surrounding this aside w/ `&mdash;`es would be better than commas.

File: draft/oss-support/post.md [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+commercial entities, is the fact that many secure services are maintained by a
+hand full of developers. These secure services are really at the front lines in
+defending privacy on the net.  Awesome as these devs are, there is only so much
+a few people can do. Bugs exists and no amount of coverage testing is going to
+find them _all_.  Similarly, take for example the [HeartBleed][link_heartbleed]
```
Comment (jerodsanto) [inline draft/oss-support/post.md]: > Similarly, take for example the [HeartBleed][link_heartbleed] event with OpenSSL.

This sentence could use a second effort.

File: draft/oss-support/post.md [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+event with OpenSSL. It surprised many to learn of the state that code base
+was in and the lack of maintainers that library had. So much so that the
```
Comment (jerodsanto) [inline draft/oss-support/post.md]: > the OpenBSD decided

Maybe add _team_ or _project_ in there?

> the OpenBSD _team_ decided

**Comment by jerodsanto** [comment]
@bechamp Sorry for taking so long to get you feedback on this. Overall, good! I left you a bunch of little notes. Hopefully you didn't get an email for each of those!

File: draft/oss-support/post.md (hunk 1 above)
Comment (bechamp) [inline draft/oss-support/post.md]: Thanks! I trusted `$ aspell -c post.md` on that one.

File: draft/oss-support/post.md (hunk 2 above)
Comment (bechamp) [inline draft/oss-support/post.md]: - Backbone Infrastructure, What's That???
  - What we take for granted.
  - I really should click that `donate` button sometime...
  - OpenSource Everywhere

I'll try harder in the morning.

**Comment by bechamp** [comment]
@jerodsanto Thanks for all of the reviews. I have local updates. Will post soon, need to find a better name. 

Ya, I got an email for each, which is fine with me. Hope this doesn't wake you up...

**Comment by bechamp** [comment]
@jerodsanto Updated my fork. I did not change the branch name to match my new proposed title but the `H1` in `post.md` shows it.

File: draft/oss-support/post.md [hunk 13]
```diff
@@ -0,0 +1,82 @@
+# Are We Taking OpenSource For Granted?
```
Comment (jerodsanto) [inline draft/oss-support/post.md]: I like this much better, but let's go w/ "Open Source" instead of "OpenSource"

File: draft/oss-support/post.md [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+
+Free software. It's not _really_ free. Time, effort, innovation and labor go
+into developing the open source projects that we use and sometimes can't live
```
Comment (jerodsanto) [inline draft/oss-support/post.md]: Instead of:

> that we use and sometimes can't live without

I would suggest:

> that we use and often rely upon

File: draft/oss-support/post.md [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+without. A lot of these projects live in the background. Many are known only to
+those that look under the hood. But these code bases are the building blocks
+of the net; the "dependencies" in our apps and OSs (see the EULA of OSX and
```
Comment (jerodsanto) [inline draft/oss-support/post.md]: s/OSX/OS X/

File: draft/oss-support/post.md [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+you'll see open source projects listed). Many open source projects find
+funding through sales of support, licenses, and by hosting. But quite a few are
```
Comment (jerodsanto) [inline draft/oss-support/post.md]: > Many open source projects find funding through sales of support, licenses, and by hosting.

Many? Maybe "Some" is more accurate? Also a bit of a rewording at the end:

> through sales of support, licenses, and/or hosting.

File: draft/oss-support/post.md [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+just put out there and sometimes they find their way into the tool boxes of
+devs and designers, to the point where they almost become ubiquitous.
```
Comment (jerodsanto) [inline draft/oss-support/post.md]: > they almost become ubiquitous.

This is a hedge, amirite?

File: draft/oss-support/post.md [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+
+When you think about it, how many of the open source projects that you use all
+the time are not heavily funded operations? VI, Emacs, heck even tools like
+[sed][link_sed] to niche languages like [Tcl][link_tcl] or [Lua][link_lua].
```
Comment (jerodsanto) [inline draft/oss-support/post.md]: > VI, Emacs, heck even tools like sed to niche languages like Tcl or Lua

This is not a complete sentence. Maybe rework this a bit? Could combine it somehow with the previous one.

File: draft/oss-support/post.md [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+These are all entrenched and still active. They may receive funding through the
+Free Software Foundation and donations, but when was the last time you paid any
```
Comment (jerodsanto) [inline draft/oss-support/post.md]: I would link up the FSF here.

**Comment by jerodsanto** [comment]
@bechamp getting close! Left you some more <strike>emails</strike> notes. After this round we'll get it in to our CMS.

Once we get it we'll want to intro you as a guest author and link to where people can connect with/follow you online. Please provide some copy for this and I'll tweak it to fit.

[Here's an example intro from a recent guest post](https://thechangelog.com/why-programming-for-kids/).

**Comment by bechamp** [comment]
@jerodsanto <https://github.com/jerodsanto>, thanks! My family and I are in
the middle of moving this week through next week. But will do my updates
asap.

On a related note, I really appreciate the time you took for me on
this and it's inspiring some more ideas in the domain of supporting open
source devs.

**Comment by bechamp** [comment]
Things are easing up between work and the move. I'll get this done soon!

**Comment by bechamp** [comment]
Changes in!

**Comment by jerodsanto** [comment]
@bechamp lookin' good! Can you provide these things?

> Once we get it we'll want to intro you as a guest author and link to where people can connect with/follow you online. Please provide some copy for this and I'll tweak it to fit.

Then we'll get this baby published!

**Comment by bechamp** [comment]
Hey @jerodsanto, 
Here is an intro:

I am a SoC Design Engineer with a focus on design automation and providing low-power solutions. I code [C](http://www.open-std.org/jtc1/sc22/wg14) and [Python](https://www.python.org); I use Open Source all the time ([Arch](https://www.archlinux.org), [Bash](http://savannah.gnu.org/projects/bash/), [dwm](http://dwm.suckless.org), [Vim](http://www.vim.org), [Git](http://www.git-scm.com), and [Firefox](https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/Source_Code/Downloading_Source_Archives) to name a few). Open Source has taught me a lot of what I know about programming and technology. I am thankful to the community and hope that Open Source's model of transparency remains successful. I currently live in Texas with my wife and daughter.

Twitter: https://twitter.com/kbechamp
GitHub: https://github.com/bechamp

**Comment by jerodsanto** [comment]
:+1: we'll get this published early next week
